### ISSUE TITLE
Randomly stops doing anything

### ISSUE THREAD

**Issue by sezanzeb** [influxdata/telegraf#13648]
### Relevant telegraf.conf

```toml
[agent]
	interval = "1s"
	flush_interval = "1s"

	debug = true
	quiet = false

[[inputs.mqtt_consumer]]
	servers = [ "mqtt://broker:1883" ]
	topics = [ "#" ]

	data_format = "json"
	tag_keys = [ "device" ]
	json_time_key = "time"
	json_time_format = "2006-01-02T15:04:05.000Z"

	[[inputs.mqtt_consumer.topic_parsing]]
		topic = "foo/+/+"
		measurement = "_/_/measurement"
		tags = "_/foo_serial/_"

    #[[inputs.mqtt_consumer.json_v2]]
    #	timestamp_path = "time"
    #	timestamp_format = "2006-01-02T15:04:05.000Z"
    #	[[inputs.mqtt_consumer.json_v2.tag]]
    #		path = "device"
	#	[[inputs.mqtt_consumer.json_v2.field]]
	#		path = "power1"
	#		type = "float"


[[processors.unpivot]]
	tag_key = "field"
	value_key = "value"

[[outputs.file]]
	files = ["stdout"]

[[outputs.postgresql]]
	connection = "host=database user=postgres password=password dbname=bar"

	tag_table_add_column_templates = []
	tag_table_create_templates = []
	add_column_templates = []
	create_templates = []

	tagexclude = [ "topic", "host" ]
```

### Logs from Telegraf

```text
telegraf-1           | 2023-07-20T07:12:33Z D! [outputs.postgresql] Wrote batch of 2 metrics in 7.919442ms
telegraf-1           | 2023-07-20T07:12:33Z D! [outputs.postgresql] Buffer fullness: 0 / 10000 metrics
telegraf-1           | 2023-07-20T07:12:34Z D! [outputs.file] Wrote batch of 2 metrics in 59.361µs
telegraf-1           | 2023-07-20T07:12:34Z D! [outputs.file] Buffer fullness: 0 / 10000 metrics
telegraf-1           | telemetry,device=foo-el-1,field=power1,host=3d36e81418bf,foo_serial=simulated,topic=foo/simulated/telemetry value=2327.1 1689837153178000000
telegraf-1           | telemetry,device=foo-el-1,field=power2,host=3d36e81418bf,foo_serial=simulated,topic=foo/simulated/telemetry value=-3658.7 1689837153178000000
```

Has been writing stuff up to this point for a few minutes every second.

From here on it just keeps repeating that the buffer is empty

```text
telegraf-1           | 2023-07-20T07:12:34Z D! [outputs.postgresql] Buffer fullness: 0 / 10000 metrics
telegraf-1           | 2023-07-20T07:12:35Z D! [outputs.file] Buffer fullness: 0 / 10000 metrics
foo-simulator-1  | [Nest] 61  - 07/20/2023, 7:12:35 AM     LOG [SimulatorService] Publishing: {"time":"2023-07-20T07:12:35.179Z","foo_serial":"simulated","device":"foo-el-1","power1":2204.6,"power2":-3189.8}
telegraf-1           | 2023-07-20T07:12:35Z D! [outputs.postgresql] Wrote batch of 2 metrics in 2.297379ms
telegraf-1           | 2023-07-20T07:12:35Z D! [outputs.postgresql] Buffer fullness: 0 / 10000 metrics
telegraf-1           | 2023-07-20T07:12:36Z D! [outputs.file] Wrote batch of 2 metrics in 60.203µs
telegraf-1           | 2023-07-20T07:12:36Z D! [outputs.file] Buffer fullness: 0 / 10000 metrics
telegraf-1           | telemetry,device=foo-el-1,field=power1,host=3d36e81418bf,foo_serial=simulated,topic=foo/simulated/telemetry value=2204.6 1689837155179000000
telegraf-1           | telemetry,device=foo-el-1,field=power2,host=3d36e81418bf,foo_serial=simulated,topic=foo/simulated/telemetry value=-3189.8 1689837155179000000
telegraf-1           | 2023-07-20T07:12:36Z D! [outputs.postgresql] Buffer fullness: 0 / 10000 metrics
telegraf-1           | 2023-07-20T07:12:37Z D! [outputs.file] Buffer fullness: 0 / 10000 metrics
foo-simulator-1  | [Nest] 61  - 07/20/2023, 7:12:37 AM     LOG [SimulatorService] Publishing: {"time":"2023-07-20T07:12:37.179Z","foo_serial":"simulated","device":"foo-el-1","power1":2666,"power2":-3899.9}
telegraf-1           | 2023-07-20T07:12:37Z D! [outputs.postgresql] Buffer fullness: 0 / 10000 metrics
telegraf-1           | 2023-07-20T07:12:38Z D! [outputs.file] Buffer fullness: 0 / 10000 metrics
telegraf-1           | 2023-07-20T07:12:38Z D! [outputs.postgresql] Buffer fullness: 0 / 10000 metrics
telegraf-1           | 2023-07-20T07:12:39Z D! [outputs.file] Buffer fullness: 0 / 10000 metrics
foo-simulator-1  | [Nest] 61  - 07/20/2023, 7:12:39 AM     LOG [SimulatorService] Publishing: {"time":"2023-07-20T07:12:39.181Z","foo_serial":"simulated","device":"foo-el-1","power1":2448,"power2":-3439.4}
telegraf-1           | 2023-07-20T07:12:39Z D! [outputs.postgresql] Buffer fullness: 0 / 10000 metrics
telegraf-1           | 2023-07-20T07:12:40Z D! [outputs.file] Buffer fullness: 0 / 10000 metrics
telegraf-1           | 2023-07-20T07:12:40Z D! [outputs.postgresql] Buffer fullness: 0 / 10000 metrics
telegraf-1           | 2023-07-20T07:12:41Z D! [outputs.file] Buffer fullness: 0 / 10000 metrics
foo-simulator-1  | [Nest] 61  - 07/20/2023, 7:12:41 AM     LOG [SimulatorService] Publishing: {"time":"2023-07-20T07:12:41.182Z","foo_serial":"simulated","device":"foo-el-1","power1":2375.7,"power2":-4005.5}
telegraf-1           | 2023-07-20T07:12:41Z D! [outputs.postgresql] Buffer fullness: 0 / 10000 metrics
telegraf-1           | 2023-07-20T07:12:42Z D! [outputs.file] Buffer fullness: 0 / 10000 metrics
telegraf-1           | 2023-07-20T07:12:42Z D! [outputs.postgresql] Buffer fullness: 0 / 10000 metrics
telegraf-1           | 2023-07-20T07:12:43Z D! [outputs.file] Buffer fullness: 0 / 10000 metrics
foo-simulator-1  | [Nest] 61  - 07/20/2023, 7:12:43 AM     LOG [SimulatorService] Publishing: {"time":"2023-07-20T07:12:43.185Z","foo_serial":"simulated","device":"foo-el-1","power1":2483.6,"power2":-3497.3}
telegraf-1           | 2023-07-20T07:12:43Z D! [outputs.postgresql] Buffer fullness: 0 / 10000 metrics
telegraf-1           | 2023-07-20T07:12:44Z D! [outputs.file] Buffer fullness: 0 / 10000 metrics
telegraf-1           | 2023-07-20T07:12:44Z D! [outputs.postgresql] Buffer fullness: 0 / 10000 metrics
telegraf-1           | 2023-07-20T07:12:45Z D! [outputs.file] Buffer fullness: 0 / 10000 metrics
foo-simulator-1  | [Nest] 61  - 07/20/2023, 7:12:45 AM     LOG [SimulatorService] Publishing: {"time":"2023-07-20T07:12:45.186Z","foo_serial":"simulated","device":"foo-el-1","power1":1980.1,"power2":-4123.9}
telegraf-1           | 2023-07-20T07:12:45Z D! [outputs.postgresql] Buffer fullness: 0 / 10000 metrics
```

### System info

Telegraf 1.27.2 (git: HEAD@9356fd83); Docker version 23.0.4, build f480fb1e37

### Docker

    telegraf:
        image: telegraf
        deploy:
            restart_policy:
                condition: on-failure
        volumes:
            - ./telegraf.conf:/etc/telegraf/telegraf.conf:ro
        depends_on:
            - database
            - broker

### Steps to reproduce

I just run the above configuration and start pulishing messages

### Expected behavior

To not stop doing stuff

### Actual behavior

Won't do anything until I restart the container (`docker-compose restart telegraf`)

**Comment by powersj** [comment]
Can you please provide your full logs, not just a snippet of them.

Are you also 100% certain you are running 1.27.2? As there was a big fix for mqtt_consumer put into that version.

**Comment by sezanzeb** [comment]
```
telegraf-1  | 2023-07-24T06:53:34Z I! Loading config: /etc/telegraf/telegraf.conf
telegraf-1  | 2023-07-24T06:53:34Z I! Starting Telegraf 1.27.2
telegraf-1  | 2023-07-24T06:53:34Z I! Available plugins: 237 inputs, 9 aggregators, 28 processors, 23 parsers, 59 outputs, 4 secret-stores
telegraf-1  | 2023-07-24T06:53:34Z I! Loaded inputs: mqtt_consumer
telegraf-1  | 2023-07-24T06:53:34Z I! Loaded aggregators: 
telegraf-1  | 2023-07-24T06:53:34Z I! Loaded processors: unpivot
telegraf-1  | 2023-07-24T06:53:34Z I! Loaded secretstores: 
telegraf-1  | 2023-07-24T06:53:34Z I! Loaded outputs: file postgresql
telegraf-1  | 2023-07-24T06:53:34Z I! Tags enabled: host=33b46a1b5e08
telegraf-1  | 2023-07-24T06:53:34Z I! [agent] Config: Interval:1s, Quiet:false, Hostname:"33b46a1b5e08", Flush Interval:1s
telegraf-1  | 2023-07-24T06:53:34Z D! [agent] Initializing plugins
telegraf-1  | 2023-07-24T06:53:34Z D! [agent] Connecting outputs
telegraf-1  | 2023-07-24T06:53:34Z D! [agent] Attempting connection to [outputs.file]
telegraf-1  | 2023-07-24T06:53:34Z D! [agent] Successfully connected to outputs.file
telegraf-1  | 2023-07-24T06:53:34Z D! [agent] Attempting connection to [outputs.postgresql]
telegraf-1  | 2023-07-24T06:53:34Z E! [outputs.postgresql] PG connect failed - map[err:failed to connect to `host=database user=postgres database=bar`: dial error (dial tcp 172.21.0.4:5432: connect: connection refused)]
telegraf-1  | 2023-07-24T06:53:34Z E! [outputs.postgresql] Couldn't connect to server
telegraf-1  | failed to connect to `host=database user=postgres database=bar`: dial error (dial tcp 172.21.0.4:5432: connect: connection refused)
telegraf-1  | 2023-07-24T06:53:34Z E! [agent] Failed to connect to [outputs.postgresql], retrying in 15s, error was "failed to connect to `host=database user=postgres database=bar`: dial error (dial tcp 172.21.0.4:5432: connect: connection refused)"
telegraf-1  | 2023-07-24T06:53:49Z D! [agent] Successfully connected to outputs.postgresql
telegraf-1  | 2023-07-24T06:53:49Z D! [agent] Starting service inputs
telegraf-1  | 2023-07-24T06:53:49Z I! [inputs.mqtt_consumer] Connected [mqtt://broker:1883]
telegraf-1  | 2023-07-24T06:53:50Z D! [outputs.postgresql] Buffer fullness: 0 / 10000 metrics
telegraf-1  | 2023-07-24T06:53:50Z D! [outputs.file] Buffer fullness: 0 / 10000 metrics
telegraf-1  | 2023-07-24T06:53:51Z D! [outputs.file] Wrote batch of 2 metrics in 124.573µs
telegraf-1  | 2023-07-24T06:53:51Z D! [outputs.file] Buffer fullness: 0 / 10000 metrics
telegraf-1  | telemetry,device=foo-el-1,field=power1,host=33b46a1b5e08,foo_serial=simulated,topic=foo/simulated/telemetry value=179 1690181631025000000
telegraf-1  | telemetry,device=foo-el-1,field=power2,host=33b46a1b5e08,foo_serial=simulated,topic=foo/simulated/telemetry value=-333.8 1690181631025000000
telegraf-1  | 2023-07-24T06:53:51Z D! [outputs.postgresql] Wrote batch of 2 metrics in 22.104766ms
telegraf-1  | 2023-07-24T06:53:51Z D! [outputs.postgresql] Buffer fullness: 0 / 10000 metrics
telegraf-1  | 2023-07-24T06:53:52Z D! [outputs.file] Buffer fullness: 0 / 10000 metrics
telegraf-1  | 2023-07-24T06:53:52Z D! [outputs.postgresql] Buffer fullness: 0 / 10000 metrics
telegraf-1  | 2023-07-24T06:53:53Z D! [outputs.file] Wrote batch of 2 metrics in 73.979µs
telegraf-1  | telemetry,device=foo-el-1,field=power1,host=33b46a1b5e08,foo_serial=simulated,topic=foo/simulated/telemetry value=220.1 1690181633028000000
telegraf-1  | telemetry,device=foo-el-1,field=power2,host=33b46a1b5e08,foo_serial=simulated,topic=foo/simulated/telemetry value=-349 1690181633028000000
telegraf-1  | 2023-07-24T06:53:53Z D! [outputs.file] Buffer fullness: 0 / 10000 metrics
telegraf-1  | 2023-07-24T06:53:53Z D! [outputs.postgresql] Wrote batch of 2 metrics in 8.71266ms
telegraf-1  | 2023-07-24T06:53:53Z D! [outputs.postgresql] Buffer fullness: 0 / 10000 metrics
telegraf-1  | 2023-07-24T06:53:54Z D! [outputs.file] Buffer fullness: 0 / 10000 metrics
telegraf-1  | 2023-07-24T06:53:54Z D! [outputs.postgresql] Buffer fullness: 0 / 10000 metrics
telegraf-1  | 2023-07-24T06:53:55Z D! [outputs.file] Wrote batch of 2 metrics in 59.502µs
telegraf-1  | 2023-07-24T06:53:55Z D! [outputs.file] Buffer fullness: 0 / 10000 metrics
telegraf-1  | telemetry,device=foo-el-1,field=power1,host=33b46a1b5e08,foo_serial=simulated,topic=foo/simulated/telemetry value=192.8 1690181635031000000
telegraf-1  | telemetry,device=foo-el-1,field=power2,host=33b46a1b5e08,foo_serial=simulated,topic=foo/simulated/telemetry value=-326.6 1690181635031000000
```

**Comment by powersj** [comment]
Hi,

Have not forgotten about this. 

What I need to see are the mqtt consumer client logs. I have added a new option to master that will allow you to see these when enabled + debug mode is enabled in the agent.

Can you grab the artifacts on https://github.com/influxdata/telegraf/pull/13663 and try them out with the `client_trace = true` option + debug enabled and provide those logs please?

Thanks

**Comment by sezanzeb** [comment]
Thanks. I'll try to get the artifacts into the docker container when I'm back. I'm away for 3 weeks.

I'm also getting this log half a minute after telegraf stopped working:

```
broker-1             | 2023-07-28 12:50:09.274310+00:00 [info] <0.783.0> MQTT connection <<"172.25.0.9:57186 -> 172.25.0.4:1883">> will terminate because peer closed TCP connection
```

Other services are still getting messages via mqtt, so the broker seems to be working.

**Comment by bjornol** [comment]
Im experiencing the same thing. Silently just stops doing anything after a while

**Comment by sezanzeb** [comment]
before and after that it just repeats forever. The terminal doesn't have the complete scrollback, if you want more I'll have to repeat it and pipe the output into a file or something.

```
foo-simulator_1  | [Nest] 61  - 08/25/2023, 7:42:39 AM     LOG [SimulatorService] Publishing: {"time":"2023-08-25T07:42:39.105Z","foo_serial":"simulated","device":"foo-el-1","power1":1966.5,"power2":-2791.1}
telegraf_1           | 2023-08-25T07:42:39Z D! [inputs.mqtt_consumer] [net]     startIncoming Received Message
telegraf_1           | 2023-08-25T07:42:39Z D! [inputs.mqtt_consumer] [net]     startIncomingComms: got msg on ibound
telegraf_1           | 2023-08-25T07:42:39Z D! [inputs.mqtt_consumer] [net]     startIncomingComms: received publish, msgId:0
telegraf_1           | 2023-08-25T07:42:39Z D! [inputs.mqtt_consumer] [net]     logic waiting for msg on ibound
telegraf_1           | 2023-08-25T07:42:39Z D! [inputs.mqtt_consumer] [pinger]  ping check44.999292471
telegraf_1           | 2023-08-25T07:42:40Z D! [outputs.file] Wrote batch of 2 metrics in 59.622µs
telegraf_1           | 2023-08-25T07:42:40Z D! [outputs.file] Buffer fullness: 0 / 10000 metrics
telegraf_1           | telemetry,device=foo-el-1,field=power1,host=e40627e14ca6,foo_serial=simulated,topic=foo/simulated/telemetry value=1966.5 1692949359105000000
telegraf_1           | telemetry,device=foo-el-1,field=power2,host=e40627e14ca6,foo_serial=simulated,topic=foo/simulated/telemetry value=-2791.1 1692949359105000000
telegraf_1           | 2023-08-25T07:42:40Z D! [outputs.postgresql] Wrote batch of 2 metrics in 8.652864ms
telegraf_1           | 2023-08-25T07:42:40Z D! [outputs.postgresql] Buffer fullness: 0 / 10000 metrics
telegraf_1           | 2023-08-25T07:42:41Z D! [outputs.file] Buffer fullness: 0 / 10000 metrics
telegraf_1           | 2023-08-25T07:42:41Z D! [outputs.postgresql] Buffer fullness: 0 / 10000 metrics
foo-simulator_1  | [Nest] 61  - 08/25/2023, 7:42:41 AM     LOG [SimulatorService] Publishing: {"time":"2023-08-25T07:42:41.106Z","foo_serial":"simulated","device":"foo-el-1","power1":2461.9,"power2":-3237.8}
telegraf_1           | 2023-08-25T07:42:41Z D! [inputs.mqtt_consumer] [net]     startIncoming Received Message
telegraf_1           | 2023-08-25T07:42:41Z D! [inputs.mqtt_consumer] [net]     startIncomingComms: got msg on ibound
telegraf_1           | 2023-08-25T07:42:41Z D! [inputs.mqtt_consumer] [net]     startIncomingComms: received publish, msgId:0
telegraf_1           | 2023-08-25T07:42:41Z D! [inputs.mqtt_consumer] [net]     logic waiting for msg on ibound
telegraf_1           | 2023-08-25T07:42:42Z D! [outputs.file] Wrote batch of 2 metrics in 56.105µs
telegraf_1           | 2023-08-25T07:42:42Z D! [outputs.file] Buffer fullness: 0 / 10000 metrics
telegraf_1           | telemetry,device=foo-el-1,field=power1,host=e40627e14ca6,foo_serial=simulated,topic=foo/simulated/telemetry value=2461.9 1692949361106000000
telegraf_1           | telemetry,device=foo-el-1,field=power2,host=e40627e14ca6,foo_serial=simulated,topic=foo/simulated/telemetry value=-3237.8 1692949361106000000
telegraf_1           | 2023-08-25T07:42:42Z D! [outputs.postgresql] Wrote batch of 2 metrics in 7.86343ms
telegraf_1           | 2023-08-25T07:42:42Z D! [outputs.postgresql] Buffer fullness: 0 / 10000 metrics
telegraf_1           | 2023-08-25T07:42:43Z D! [outputs.file] Buffer fullness: 0 / 10000 metrics
telegraf_1           | 2023-08-25T07:42:43Z D! [outputs.postgresql] Buffer fullness: 0 / 10000 metrics
foo-simulator_1  | [Nest] 61  - 08/25/2023, 7:42:43 AM     LOG [SimulatorService] Publishing: {"time":"2023-08-25T07:42:43.108Z","foo_serial":"simulated","device":"foo-el-1","power1":2306,"power2":-2826.4}
telegraf_1           | 2023-08-25T07:42:43Z D! [inputs.mqtt_consumer] [net]     startIncoming Received Message
telegraf_1           | 2023-08-25T07:42:43Z D! [inputs.mqtt_consumer] [net]     startIncomingComms: got msg on ibound
telegraf_1           | 2023-08-25T07:42:43Z D! [inputs.mqtt_consumer] [net]     startIncomingComms: received publish, msgId:0
telegraf_1           | 2023-08-25T07:42:43Z D! [inputs.mqtt_consumer] [net]     logic waiting for msg on ibound
telegraf_1           | 2023-08-25T07:42:44Z D! [outputs.file] Buffer fullness: 0 / 10000 metrics
telegraf_1           | 2023-08-25T07:42:44Z D! [outputs.postgresql] Buffer fullness: 0 / 10000 metrics
telegraf_1           | 2023-08-25T07:42:44Z D! [inputs.mqtt_consumer] [pinger]  ping check49.999309158
telegraf_1           | 2023-08-25T07:42:45Z D! [outputs.file] Buffer fullness: 0 / 10000 metrics
telegraf_1           | 2023-08-25T07:42:45Z D! [outputs.postgresql] Buffer fullness: 0 / 10000 metrics
foo-simulator_1  | [Nest] 61  - 08/25/2023, 7:42:45 AM     LOG [SimulatorService] Publishing: {"time":"2023-08-25T07:42:45.110Z","foo_serial":"simulated","device":"foo-el-1","power1":2300.4,"power2":-2992.5}
telegraf_1           | 2023-08-25T07:42:45Z D! [inputs.mqtt_consumer] [net]     startIncoming Received Message
telegraf_1           | 2023-08-25T07:42:45Z D! [inputs.mqtt_consumer] [net]     startIncomingComms: got msg on ibound
telegraf_1           | 2023-08-25T07:42:45Z D! [inputs.mqtt_consumer] [net]     startIncomingComms: received publish, msgId:0
telegraf_1           | 2023-08-25T07:42:45Z D! [inputs.mqtt_consumer] [net]     logic waiting for msg on ibound
telegraf_1           | 2023-08-25T07:42:46Z D! [outputs.file] Buffer fullness: 0 / 10000 metrics
telegraf_1           | 2023-08-25T07:42:46Z D! [outputs.postgresql] Buffer fullness: 0 / 10000 metrics
telegraf_1           | 2023-08-25T07:42:47Z D! [outputs.file] Buffer fullness: 0 / 10000 metrics
telegraf_1           | 2023-08-25T07:42:47Z D! [outputs.postgresql] Buffer fullness: 0 / 10000 metrics
foo-simulator_1  | [Nest] 61  - 08/25/2023, 7:42:47 AM     LOG [SimulatorService] Publishing: {"time":"2023-08-25T07:42:47.112Z","foo_serial":"simulated","device":"foo-el-1","power1":2650.4,"power2":-3043.7}
telegraf_1           | 2023-08-25T07:42:47Z D! [inputs.mqtt_consumer] [net]     startIncoming Received Message
telegraf_1           | 2023-08-25T07:42:47Z D! [inputs.mqtt_consumer] [net]     startIncomingComms: got msg on ibound
telegraf_1           | 2023-08-25T07:42:47Z D! [inputs.mqtt_consumer] [net]     startIncomingComms: received publish, msgId:0
telegraf_1           | 2023-08-25T07:42:47Z D! [inputs.mqtt_consumer] [net]     logic waiting for msg on ibound
telegraf_1           | 2023-08-25T07:42:48Z D! [outputs.file] Buffer fullness: 0 / 10000 metrics
telegraf_1           | 2023-08-25T07:42:48Z D! [outputs.postgresql] Buffer fullness: 0 / 10000 metrics
telegraf_1           | 2023-08-25T07:42:49Z D! [outputs.file] Buffer fullness: 0 / 10000 metrics
telegraf_1           | 2023-08-25T07:42:49Z D! [outputs.postgresql] Buffer fullness: 0 / 10000 metrics
foo-simulator_1  | [Nest] 61  - 08/25/2023, 7:42:49 AM     LOG [SimulatorService] Publishing: {"time":"2023-08-25T07:42:49.114Z","foo_serial":"simulated","device":"foo-el-1","power1":2301.1,"power2":-3763.8}
telegraf_1           | 2023-08-25T07:42:49Z D! [inputs.mqtt_consumer] [net]     startIncoming Received Message
telegraf_1           | 2023-08-25T07:42:49Z D! [inputs.mqtt_consumer] [net]     startIncomingComms: got msg on ibound
telegraf_1           | 2023-08-25T07:42:49Z D! [inputs.mqtt_consumer] [net]     startIncomingComms: received publish, msgId:0
telegraf_1           | 2023-08-25T07:42:49Z D! [inputs.mqtt_consumer] [pinger]  ping check54.999824252
telegraf_1           | 2023-08-25T07:42:50Z D! [outputs.file] Buffer fullness: 0 / 10000 metrics
telegraf_1           | 2023-08-25T07:42:50Z D! [outputs.postgresql] Buffer fullness: 0 / 10000 metrics
telegraf_1           | 2023-08-25T07:42:51Z D! [outputs.file] Buffer fullness: 0 / 10000 metrics
telegraf_1           | 2023-08-25T07:42:51Z D! [outputs.postgresql] Buffer fullness: 0 / 10000 metrics
foo-simulator_1  | [Nest] 61  - 08/25/2023, 7:42:51 AM     LOG [SimulatorService] Publishing: {"time":"2023-08-25T07:42:51.117Z","foo_serial":"simulated","device":"foo-el-1","power1":2359.9,"power2":-3810.4}
telegraf_1           | 2023-08-25T07:42:51Z D! [inputs.mqtt_consumer] [net]     startIncoming Received Message
telegraf_1           | 2023-08-25T07:42:52Z D! [outputs.file] Buffer fullness: 0 / 10000 metrics
telegraf_1           | 2023-08-25T07:42:52Z D! [outputs.postgresql] Buffer fullness: 0 / 10000 metrics
telegraf_1           | 2023-08-25T07:42:53Z D! [outputs.file] Buffer fullness: 0 / 10000 metrics
telegraf_1           | 2023-08-25T07:42:53Z D! [outputs.postgresql] Buffer fullness: 0 / 10000 metrics
foo-simulator_1  | [Nest] 61  - 08/25/2023, 7:42:53 AM     LOG [SimulatorService] Publishing: {"time":"2023-08-25T07:42:53.118Z","foo_serial":"simulated","device":"foo-el-1","power1":2495.2,"power2":-3726.9}
telegraf_1           | 2023-08-25T07:42:54Z D! [outputs.file] Buffer fullness: 0 / 10000 metrics
telegraf_1           | 2023-08-25T07:42:54Z D! [outputs.postgresql] Buffer fullness: 0 / 10000 metrics
telegraf_1           | 2023-08-25T07:42:54Z D! [inputs.mqtt_consumer] [pinger]  ping check59.999517339
telegraf_1           | 2023-08-25T07:42:55Z D! [outputs.file] Buffer fullness: 0 / 10000 metrics
telegraf_1           | 2023-08-25T07:42:55Z D! [outputs.postgresql] Buffer fullness: 0 / 10000 metrics
foo-simulator_1  | [Nest] 61  - 08/25/2023, 7:42:55 AM     LOG [SimulatorService] Publishing: {"time":"2023-08-25T07:42:55.120Z","foo_serial":"simulated","device":"foo-el-1","power1":2705.3,"power2":-4001.8}
telegraf_1           | 2023-08-25T07:42:56Z D! [outputs.file] Buffer fullness: 0 / 10000 metrics
telegraf_1           | 2023-08-25T07:42:56Z D! [outputs.postgresql] Buffer fullness: 0 / 10000 metrics
telegraf_1           | 2023-08-25T07:42:57Z D! [outputs.file] Buffer fullness: 0 / 10000 metrics
telegraf_1           | 2023-08-25T07:42:57Z D! [outputs.postgresql] Buffer fullness: 0 / 10000 metrics
foo-simulator_1  | [Nest] 61  - 08/25/2023, 7:42:57 AM     LOG [SimulatorService] Publishing: {"time":"2023-08-25T07:42:57.122Z","foo_serial":"simulated","device":"foo-el-1","power1":2209,"power2":-3748.6}
telegraf_1           | 2023-08-25T07:42:58Z D! [outputs.file] Buffer fullness: 0 / 10000 metrics
telegraf_1           | 2023-08-25T07:42:58Z D! [outputs.postgresql] Buffer fullness: 0 / 10000 metrics
telegraf_1           | 2023-08-25T07:42:59Z D! [outputs.file] Buffer fullness: 0 / 10000 metrics
telegraf_1           | 2023-08-25T07:42:59Z D! [outputs.postgresql] Buffer fullness: 0 / 10000 metrics
foo-simulator_1  | [Nest] 61  - 08/25/2023, 7:42:59 AM     LOG [SimulatorService] Publishing: {"time":"2023-08-25T07:42:59.125Z","foo_serial":"simulated","device":"foo-el-1","power1":2284.2,"power2":-2984.9}
telegraf_1           | 2023-08-25T07:42:59Z D! [inputs.mqtt_consumer] [pinger]  ping check64.999533344
telegraf_1           | 2023-08-25T07:42:59Z D! [inputs.mqtt_consumer] [pinger]  keepalive sending ping
telegraf_1           | 2023-08-25T07:43:00Z D! [outputs.file] Buffer fullness: 0 / 10000 metrics
telegraf_1           | 2023-08-25T07:43:00Z D! [outputs.postgresql] Buffer fullness: 0 / 10000 metrics
telegraf_1           | 2023-08-25T07:43:01Z D! [outputs.file] Buffer fullness: 0 / 10000 metrics
telegraf_1           | 2023-08-25T07:43:01Z D! [outputs.postgresql] Buffer fullness: 0 / 10000 metrics
foo-simulator_1  | [Nest] 61  - 08/25/2023, 7:43:01 AM     LOG [SimulatorService] Publishing: {"time":"2023-08-25T07:43:01.126Z","foo_serial":"simulated","device":"foo-el-1","power1":2751,"power2":-3901.2}
telegraf_1           | 2023-08-25T07:43:02Z D! [outputs.file] Buffer fullness: 0 / 10000 metrics
telegraf_1           | 2023-08-25T07:43:02Z D! [outputs.postgresql] Buffer fullness: 0 / 10000 metrics
telegraf_1           | 2023-08-25T07:43:03Z D! [outputs.file] Buffer fullness: 0 / 10000 metrics
telegraf_1           | 2023-08-25T07:43:03Z D! [outputs.postgresql] Buffer fullness: 0 / 10000 metrics
foo-simulator_1  | [Nest] 61  - 08/25/2023, 7:43:03 AM     LOG [SimulatorService] Publishing: {"time":"2023-08-25T07:43:03.128Z","foo_serial":"simulated","device":"foo-el-1","power1":2032.1,"power2":-3568.4}
telegraf_1           | 2023-08-25T07:43:04Z D! [outputs.file] Buffer fullness: 0 / 10000 metrics
telegraf_1           | 2023-08-25T07:43:04Z D! [outputs.postgresql] Buffer fullness: 0 / 10000 metrics
telegraf_1           | 2023-08-25T07:43:04Z D! [inputs.mqtt_consumer] [pinger]  ping check4.999813113
telegraf_1           | 2023-08-25T07:43:05Z D! [outputs.file] Buffer fullness: 0 / 10000 metrics
telegraf_1           | 2023-08-25T07:43:05Z D! [outputs.postgresql] Buffer fullness: 0 / 10000 metrics
foo-simulator_1  | [Nest] 61  - 08/25/2023, 7:43:05 AM     LOG [SimulatorService] Publishing: {"time":"2023-08-25T07:43:05.131Z","foo_serial":"simulated","device":"foo-el-1","power1":2410.7,"power2":-3460.1}
telegraf_1           | 2023-08-25T07:43:06Z D! [outputs.file] Buffer fullness: 0 / 10000 metrics
telegraf_1           | 2023-08-25T07:43:06Z D! [outputs.postgresql] Buffer fullness: 0 / 10000 metrics
telegraf_1           | 2023-08-25T07:43:07Z D! [outputs.file] Buffer fullness: 0 / 10000 metrics
telegraf_1           | 2023-08-25T07:43:07Z D! [outputs.postgresql] Buffer fullness: 0 / 10000 metrics
foo-simulator_1  | [Nest] 61  - 08/25/2023, 7:43:07 AM     LOG [SimulatorService] Publishing: {"time":"2023-08-25T07:43:07.134Z","foo_serial":"simulated","device":"foo-el-1","power1":2075,"power2":-3228.7}
telegraf_1           | 2023-08-25T07:43:08Z D! [outputs.file] Buffer fullness: 0 / 10000 metrics
telegraf_1           | 2023-08-25T07:43:08Z D! [outputs.postgresql] Buffer fullness: 0 / 10000 metrics
telegraf_1           | 2023-08-25T07:43:09Z D! [outputs.file] Buffer fullness: 0 / 10000 metrics
telegraf_1           | 2023-08-25T07:43:09Z D! [outputs.postgresql] Buffer fullness: 0 / 10000 metrics
foo-simulator_1  | [Nest] 61  - 08/25/2023, 7:43:09 AM     LOG [SimulatorService] Publishing: {"time":"2023-08-25T07:43:09.136Z","foo_serial":"simulated","device":"foo-el-1","power1":2298.6,"power2":-3366.6}
telegraf_1           | 2023-08-25T07:43:09Z D! [inputs.mqtt_consumer] [pinger]  ping check10.000246964
telegraf_1           | 2023-08-25T07:43:09Z D! [inputs.mqtt_consumer] [pinger]  pingresp not received, disconnecting
telegraf_1           | 2023-08-25T07:43:09Z D! [inputs.mqtt_consumer] [client]  internalConnLost called
telegraf_1           | 2023-08-25T07:43:09Z D! [inputs.mqtt_consumer] [client]  stopCommsWorkers called
telegraf_1           | 2023-08-25T07:43:09Z D! [inputs.mqtt_consumer] [client]  internalConnLost waiting on workers
telegraf_1           | 2023-08-25T07:43:09Z D! [inputs.mqtt_consumer] [client]  stopCommsWorkers waiting for workers
broker_1             | 2023-08-25 07:43:09.844025+00:00 [info] <0.659.0> MQTT connection <<"172.20.0.9:49640 -> 172.20.0.3:1883">> will terminate because peer closed TCP connection
telegraf_1           | 2023-08-25T07:43:10Z D! [outputs.file] Buffer fullness: 0 / 10000 metrics
telegraf_1           | 2023-08-25T07:43:10Z D! [outputs.postgresql] Buffer fullness: 0 / 10000 metrics
telegraf_1           | 2023-08-25T07:43:11Z D! [outputs.file] Buffer fullness: 0 / 10000 metrics
telegraf_1           | 2023-08-25T07:43:11Z D! [outputs.postgresql] Buffer fullness: 0 / 10000 metrics
foo-simulator_1  | [Nest] 61  - 08/25/2023, 7:43:11 AM     LOG [SimulatorService] Publishing: {"time":"2023-08-25T07:43:11.138Z","foo_serial":"simulated","device":"foo-el-1","power1":2672.6,"power2":-3430.4}
telegraf_1           | 2023-08-25T07:43:12Z D! [outputs.file] Buffer fullness: 0 / 10000 metrics
telegraf_1           | 2023-08-25T07:43:12Z D! [outputs.postgresql] Buffer fullness: 0 / 10000 metrics
telegraf_1           | 2023-08-25T07:43:13Z D! [outputs.file] Buffer fullness: 0 / 10000 metrics
telegraf_1           | 2023-08-25T07:43:13Z D! [outputs.postgresql] Buffer fullness: 0 / 10000 metrics
foo-simulator_1  | [Nest] 61  - 08/25/2023, 7:43:13 AM     LOG [SimulatorService] Publishing: {"time":"2023-08-25T07:43:13.141Z","foo_serial":"simulated","device":"foo-el-1","power1":2384.5,"power2":-2900.5}
```

**Comment by sezanzeb** [comment]
Here is the startup log. The broker wasn't running at first, so It tried to restart a few times.

```
telegraf_1           | 2023-08-25T07:09:19Z I! Loading config: /etc/telegraf/telegraf.conf
telegraf_1           | 2023-08-25T07:09:19Z I! Starting Telegraf 1.27.4
telegraf_1           | 2023-08-25T07:09:19Z I! Available plugins: 237 inputs, 9 aggregators, 28 processors, 23 parsers, 59 outputs, 4 secret-stores
telegraf_1           | 2023-08-25T07:09:19Z I! Loaded inputs: mqtt_consumer
telegraf_1           | 2023-08-25T07:09:19Z I! Loaded aggregators: 
telegraf_1           | 2023-08-25T07:09:19Z I! Loaded processors: unpivot
telegraf_1           | 2023-08-25T07:09:19Z I! Loaded secretstores: 
telegraf_1           | 2023-08-25T07:09:19Z I! Loaded outputs: file postgresql
telegraf_1           | 2023-08-25T07:09:19Z I! Tags enabled: host=e40627e14ca6
telegraf_1           | 2023-08-25T07:09:19Z I! [agent] Config: Interval:1s, Quiet:false, Hostname:"e40627e14ca6", Flush Interval:1s
telegraf_1           | 2023-08-25T07:09:19Z D! [agent] Initializing plugins
telegraf_1           | 2023-08-25T07:09:19Z D! [agent] Connecting outputs
telegraf_1           | 2023-08-25T07:09:19Z D! [agent] Attempting connection to [outputs.file]
telegraf_1           | 2023-08-25T07:09:19Z D! [agent] Successfully connected to outputs.file
telegraf_1           | 2023-08-25T07:09:19Z D! [agent] Attempting connection to [outputs.postgresql]
telegraf_1           | 2023-08-25T07:09:19Z D! [agent] Successfully connected to outputs.postgresql
telegraf_1           | 2023-08-25T07:09:19Z D! [agent] Starting service inputs
telegraf_1           | 2023-08-25T07:09:19Z D! [inputs.mqtt_consumer] [client]  Connect()
telegraf_1           | 2023-08-25T07:09:19Z D! [inputs.mqtt_consumer] [store]   memorystore initialized
telegraf_1           | 2023-08-25T07:09:19Z D! [inputs.mqtt_consumer] [client]  about to write new connect msg
telegraf_1           | 2023-08-25T07:09:19Z D! [inputs.mqtt_consumer] [client]  socket connected to broker
telegraf_1           | 2023-08-25T07:09:19Z D! [inputs.mqtt_consumer] [client]  Using MQTT 3.1.1 protocol
telegraf_1           | 2023-08-25T07:09:19Z D! [inputs.mqtt_consumer] [net]     connect started
telegraf_1           | 2023-08-25T07:09:19Z D! [inputs.mqtt_consumer] [net]     received connack
telegraf_1           | 2023-08-25T07:09:19Z D! [inputs.mqtt_consumer] [client]  startCommsWorkers called
telegraf_1           | 2023-08-25T07:09:19Z D! [inputs.mqtt_consumer] [client]  client is connected/reconnected
telegraf_1           | 2023-08-25T07:09:19Z D! [inputs.mqtt_consumer] [net]     incoming started
telegraf_1           | 2023-08-25T07:09:19Z D! [inputs.mqtt_consumer] [net]     startIncomingComms started
telegraf_1           | 2023-08-25T07:09:19Z D! [inputs.mqtt_consumer] [net]     outgoing started
telegraf_1           | 2023-08-25T07:09:19Z D! [inputs.mqtt_consumer] [net]     startComms started
telegraf_1           | 2023-08-25T07:09:19Z D! [inputs.mqtt_consumer] [client]  startCommsWorkers done
telegraf_1           | 2023-08-25T07:09:19Z D! [inputs.mqtt_consumer] [pinger]  keepalive starting
telegraf_1           | 2023-08-25T07:09:19Z D! [inputs.mqtt_consumer] [net]     logic waiting for msg on ibound
telegraf_1           | 2023-08-25T07:09:19Z D! [inputs.mqtt_consumer] [store]   memorystore wiped
telegraf_1           | 2023-08-25T07:09:19Z D! [inputs.mqtt_consumer] [client]  exit startClient
telegraf_1           | 2023-08-25T07:09:19Z D! [inputs.mqtt_consumer] [net]     outgoing waiting for an outbound message
telegraf_1           | 2023-08-25T07:09:19Z D! [inputs.mqtt_consumer] [net]     startIncomingComms: inboundFromStore complete
telegraf_1           | 2023-08-25T07:09:19Z D! [inputs.mqtt_consumer] [net]     logic waiting for msg on ibound
telegraf_1           | 2023-08-25T07:09:19Z I! [inputs.mqtt_consumer] Connected [mqtt://broker:1883]
telegraf_1           | 2023-08-25T07:09:19Z D! [inputs.mqtt_consumer] [client]  enter SubscribeMultiple
telegraf_1           | 2023-08-25T07:09:19Z D! [inputs.mqtt_consumer] [client]  sending subscribe message, topics:[#]
telegraf_1           | 2023-08-25T07:09:19Z D! [inputs.mqtt_consumer] [client]  exit SubscribeMultiple
telegraf_1           | 2023-08-25T07:09:19Z D! [inputs.mqtt_consumer] [net]     obound priority msg to write, type*packets.SubscribePacket
telegraf_1           | 2023-08-25T07:09:19Z D! [inputs.mqtt_consumer] [net]     outgoing waiting for an outbound message
telegraf_1           | 2023-08-25T07:09:19Z D! [inputs.mqtt_consumer] [net]     startIncoming Received Message
telegraf_1           | 2023-08-25T07:09:19Z D! [inputs.mqtt_consumer] [net]     startIncomingComms: got msg on ibound
telegraf_1           | 2023-08-25T07:09:19Z D! [inputs.mqtt_consumer] [store]   memorystore del: message1not found
telegraf_1           | 2023-08-25T07:09:19Z D! [inputs.mqtt_consumer] [net]     startIncomingComms: received suback, id:1
telegraf_1           | 2023-08-25T07:09:19Z D! [inputs.mqtt_consumer] [net]     startIncomingComms: granted qoss[0]
telegraf_1           | 2023-08-25T07:09:19Z D! [inputs.mqtt_consumer] [net]     logic waiting for msg on ibound
telegraf_1           | 2023-08-25T07:09:20Z D! [outputs.file] Buffer fullness: 0 / 10000 metrics
telegraf_1           | 2023-08-25T07:09:20Z D! [outputs.postgresql] Buffer fullness: 0 / 10000 metrics
telegraf_1           | 2023-08-25T07:09:21Z D! [inputs.mqtt_consumer] [net]     startIncoming Received Message
telegraf_1           | 2023-08-25T07:09:21Z D! [inputs.mqtt_consumer] [net]     startIncomingComms: got msg on ibound
telegraf_1           | 2023-08-25T07:09:21Z D! [inputs.mqtt_consumer] [net]     startIncomingComms: received publish, msgId:0
telegraf_1           | 2023-08-25T07:09:21Z D! [inputs.mqtt_consumer] [net]     logic waiting for msg on ibound
telegraf_1           | telemetry,device=foo-el-1,field=power1,host=e40627e14ca6,foo_serial=simulated,topic=foo/simulated/telemetry value=103.7 1692947361081000000
telegraf_1           | telemetry,device=foo-el-1,field=power2,host=e40627e14ca6,foo_serial=simulated,topic=foo/simulated/telemetry value=-182.7 1692947361081000000
telegraf_1           | 2023-08-25T07:09:21Z D! [outputs.file] Wrote batch of 2 metrics in 111.188µs
telegraf_1           | 2023-08-25T07:09:21Z D! [outputs.file] Buffer fullness: 0 / 10000 metrics
telegraf_1           | 2023-08-25T07:09:21Z D! [outputs.postgresql] Wrote batch of 2 metrics in 20.514931ms
telegraf_1           | 2023-08-25T07:09:21Z D! [outputs.postgresql] Buffer fullness: 0 / 10000 metrics
telegraf_1           | 2023-08-25T07:09:22Z D! [outputs.postgresql] Buffer fullness: 0 / 10000 metrics
telegraf_1           | 2023-08-25T07:09:22Z D! [outputs.file] Buffer fullness: 0 / 10000 metrics
```

If you want more logs, I'll filter the complete docker-compose output of today and upload it as a file
